app/src/main/scala/uk/ac/wellcome/platformapi/modules/ElasticClientModule.scala

```diff
@@ -0,0 +1,53 @@
+package com.practice.modules
+
+import javax.inject.Singleton
+
+import com.google.inject.Provides
+import com.sksamuel.elastic4s.ElasticClient
+import com.sksamuel.elastic4s.xpack.security.XPackElasticClient
+import com.sksamuel.elastic4s.ElasticsearchClientUri
+import com.twitter.inject.TwitterModule
+import org.elasticsearch.common.settings.Settings
+
+
+case class XPackConfig(user: String, ssl: Boolean)
+
+object XPackConfigModule extends TwitterModule {
+  private val xpackEnabled = flag("xpack.enabled", false, "xpack enabled")
+  private val user = flag("xpack.user", "", "xpack user:password")
+  private val sslEnabled = flag("xpack.sslEnabled", true, "xpack use ssl")
+
+  @Singleton
+  @Provides
+  def providesXPackConfig(): Option[XPackConfig] =
+    Option(xpackEnabled()).collect { case true =>
+      XPackConfig(user(), sslEnabled())
+    }
+}
+
+object ElasticClientModule extends TwitterModule {
```

Comment:
@jamesgorrie take a look at how Finatra does dependency injection ... it's beautiful.